pkg/variantregistry/syncer.go

```diff
@@ -0,0 +1,230 @@
+	log.Infof("deleting %d job variants", len(deletes))
+	for i, jv := range deletes {
+		uLog := log.WithField("progress", fmt.Sprintf("%d/%d", i+1, len(deletes)))
+		err = deleteVariant(uLog, bqClient, jv)
+		if err != nil {
+			log.WithError(err).Error("error syncing job variants to bigquery")
+		}
+	}
+
+	// Delete jobs entirely, much faster than one variant at a time when jobs have been removed.
+	// This should be relatively rare and would require the job to not have run for weeks/months.
+	log.Infof("deleting %d jobs", len(deleteJobs))
+	for i, job := range deleteJobs {
+		uLog := log.WithField("progress", fmt.Sprintf("%d/%d", i+1, len(updates)))
+		err = deleteJob(uLog, bqClient, job)
+		if err != nil {
+			log.WithError(err).Error("error syncing job variants to bigquery")
+		}
+	}
+
+	return nil
+}
+
+// compareVariants compares the list of variants vs expected and returns the variants to be inserted, deleted, and updated.
+// Broken out for unit testing purposes.
+func compareVariants(expectedVariants, currentVariants map[string]map[string]string) (insertVariants, updateVariants, deleteVariants []jobVariant, deleteJobs []string) {
+	insertVariants = []jobVariant{}
+	updateVariants = []jobVariant{}
+	deleteVariants = []jobVariant{}
+	deleteJobs = []string{}
+
+	for expectedJob, expectedVariants := range expectedVariants {
+		if _, ok := currentVariants[expectedJob]; !ok {
+			// Handle net new jobs:
+			for k, v := range expectedVariants {
+				insertVariants = append(insertVariants, jobVariant{
+					JobName:      expectedJob,
+					VariantName:  k,
+					VariantValue: v,
+				})
+			}
+			continue
+		}
+
+		// Sync variants for an existing job if any have changed:
+		for k, v := range expectedVariants {
+			currVarVal, ok := currentVariants[expectedJob][k]
+			if !ok {
+				// New variant added:
+				insertVariants = append(insertVariants, jobVariant{
+					JobName:      expectedJob,
+					VariantName:  k,
+					VariantValue: v,
+				})
+			} else {
+				if currVarVal != v {
+					updateVariants = append(updateVariants, jobVariant{
+						JobName:      expectedJob,
+						VariantName:  k,
+						VariantValue: v,
+					})
+				}
+			}
+		}
+
+		// Look for any variants for this job that should be removed:
+		for k, v := range currentVariants[expectedJob] {
+			if _, ok := expectedVariants[k]; !ok {
+				deleteVariants = append(deleteVariants, jobVariant{
+					JobName:      expectedJob,
+					VariantName:  k,
+					VariantValue: v,
+				})
+			}
+		}
+	}
+
+	// Look for any jobs that should be removed:
+	for currJobName := range currentVariants {
+		if _, ok := expectedVariants[currJobName]; !ok {
+			deleteJobs = append(deleteJobs, currJobName)
+		}
+	}
+
+	return insertVariants, updateVariants, deleteVariants, deleteJobs
+}
+
+func loadCurrentJobVariants(bqClient *bigquery.Client) (map[string]map[string]string, error) {
+	query := bqClient.Query(`SELECT * FROM openshift-ci-data-analysis.sippy.JobVariants ORDER BY JobName, VariantName`)
```

Comment:
@stbenjam  What did we decide on that meeting I missed the end of, sippy or ci-data-analysis or gce-devel.something?